cloudify_ansible/utils.py

```diff
@@ -624,32 +742,139 @@ def get_executable_path(executable, venv):
     return '{0}/bin/{1}'.format(venv, executable) if venv else executable
 
 
-def create_playbook_venv(_ctx,
-                         packages_to_install=None,
-                         collections_to_install=None):
```

Comment:
Let's move `get_instance(_ctx)` to before the `if`, since this is used a three times in `install_new_pyenv_condition`, e.g.:

```
     if get_instance(_ctx).runtime_properties.get(PLAYBOOK_VENV):
```

```
                 get_instance(_ctx).runtime_properties.get(PLAYBOOK_VENV)
```

```

        get_instance(_ctx).runtime_properties[PLAYBOOK_VENV] = \
            _ctx.node.properties.get('ansible_external_pyenv')
```